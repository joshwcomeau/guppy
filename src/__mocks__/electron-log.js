/* eslint-disable flowtype/require-valid-file-annotation */

module.exports = {
  transports: {
    file: {
      level: 'info',
    },
  },
};
