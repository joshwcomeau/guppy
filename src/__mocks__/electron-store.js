/* eslint-disable flowtype/require-valid-file-annotation */
module.exports = function() {
  let store = {};

  this.clear = jest.fn(() => {
    store = {};
  });

  this.get = jest.fn(key => store[key]);
  this.set = jest.fn((key, val) => {
    store[key] = val;
    return val;
  });
};
