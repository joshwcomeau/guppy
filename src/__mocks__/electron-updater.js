/* eslint-disable flowtype/require-valid-file-annotation */

module.exports = {
  autoUpdater: {
    checkForUpdatesAndNotify: jest.fn(),
    on: jest.fn(),
  },
};
