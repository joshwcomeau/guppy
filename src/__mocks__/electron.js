/* eslint-disable flowtype/require-valid-file-annotation */
import * as path from 'path';

module.exports = {
  ipcRenderer: {
    send: jest.fn(),
  },
  remote: {
    app: {
      getAppPath: () => path.resolve(__dirname, '..', '..', '..'),
      getPath: () =>
        process.env.APPDATA ||
        (process.platform === 'darwin'
          ? process.env.HOME + 'Library/Preferences'
          : '/var/local'),
    },
    dialog: {
      showErrorBox: jest.fn(),
      showOpenDialog: jest.fn(),
      showMessageBox: jest.fn(),
    },
    shell: {
      moveItemToTrash: jest.fn(),
    },
  },
};
