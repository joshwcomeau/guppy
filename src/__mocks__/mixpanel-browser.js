/* eslint-disable flowtype/require-valid-file-annotation */
module.exports = {
  init: jest.fn(),
  identify: jest.fn(),
  track: jest.fn(),
  has_opted_out_tracking: jest.fn(() => false),
};
