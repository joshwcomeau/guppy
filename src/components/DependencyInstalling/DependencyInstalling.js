// @flow
import React, { Component, Fragment } from 'react';
import styled from 'styled-components';

import guppyLoaderSrc from '../../assets/images/guppy-loader.gif';
import { COLORS } from '../../constants';

import Heading from '../Heading';
import Spacer from '../Spacer';

type Props = {
  name: string,
  queued: boolean,
};

class DependencyInstalling extends Component<Props> {
  render() {
    const { name, queued } = this.props;
    const stylizedName = (
      <span style={{ color: COLORS.purple[500] }}>{name}</span>
    );

    return (
      <Wrapper>
        <InnerWrapper>
          <GuppyImage src={guppyLoaderSrc} />
          <Spacer size={50} />
          <Heading size="small">
            {queued ? (
              <Fragment>{stylizedName} is queued for install...</Fragment>
            ) : (
              <Fragment>Installing {stylizedName}...</Fragment>
            )}
          </Heading>
        </InnerWrapper>
      </Wrapper>
    );
  }
}

const Wrapper = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

const InnerWrapper = styled.div`
  text-align: center;
`;

const GuppyImage = styled.img`
  width: 148px;
`;

export default DependencyInstalling;
