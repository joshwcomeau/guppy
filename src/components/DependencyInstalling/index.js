// @flow
export { default } from './DependencyInstalling';
