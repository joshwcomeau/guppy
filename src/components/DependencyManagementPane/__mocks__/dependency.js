// @flow
export const mockReactHit = {
  name: 'React',
  version: '16.0.0',
  license: 'MIT',
  modified: 1549828927372,
  humanDownloadsLast30Days: 1000000,
};
