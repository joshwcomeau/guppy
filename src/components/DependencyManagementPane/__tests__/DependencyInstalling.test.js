import React from 'react';
import { shallow } from 'enzyme';

import DependencyInstalling from '../DependencyInstalling';
import { mockReactHit } from '../__mocks__/dependency';

describe('DependencyInstalling component', () => {
  const wrapper = shallow(<DependencyInstalling name={mockReactHit.name} />);

  it('should render (queued)', () => {
    wrapper.setProps({
      queued: true,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render (installing)', () => {
    wrapper.setProps({
      queued: false,
    });
    expect(wrapper).toMatchSnapshot();
  });
});
