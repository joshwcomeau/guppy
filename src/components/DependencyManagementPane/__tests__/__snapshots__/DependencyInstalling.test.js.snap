// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DependencyInstalling component should render (installing) 1`] = `
<styled.div>
  <styled.div>
    <styled.img
      src="guppy-loader.gif"
    />
    <styled.div
      size={50}
    />
    <Heading
      size="small"
    >
      Installing 
      <span
        style={
          Object {
            "color": "#651fff",
          }
        }
      >
        React
      </span>
      ...
    </Heading>
  </styled.div>
</styled.div>
`;

exports[`DependencyInstalling component should render (queued) 1`] = `
<styled.div>
  <styled.div>
    <styled.img
      src="guppy-loader.gif"
    />
    <styled.div
      size={50}
    />
    <Heading
      size="small"
    >
      <span
        style={
          Object {
            "color": "#651fff",
          }
        }
      >
        React
      </span>
       is queued for install...
    </Heading>
  </styled.div>
</styled.div>
`;
